base/pkg/libgit2/strarray.jl

```diff
@@ -0,0 +1,25 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+function StrArrayStruct{T<:AbstractString}(strs::T...)
+    count = length(strs)
+    strings = convert(Ptr{Ptr{Uint8}}, Libc.malloc(sizeof(Ptr{Uint8}) * count))
+    for i=1:count
+        len = length(strs[i])
+        #in_ptr  = convert(Ptr{Uint8}, bytestring(strs[i]))
+        in_ptr  = pointer(bytestring(strs[i]))
```

Comment:
Use `unsafe_convert(Cstring, strs[i])` here, in order to throw an error if the strs contain NULL chars.   (Then you have to also wrap the whole thing in a `try ... finally` construct in order to avoid memory leaks if an exception is thrown.)